### PR TITLE
[#163690184]: An incident report can be marked as newly reported

### PR DESCRIPTION
**Pull request by nelmandela**
### What does PR do?

	•Allow admin to mark incident reports as newly reported

### Description of task to be completed:

        •Add state transitions field 
	   •Add transition constants

### How can it be tested manually?

	•Navigate to incidence reports endpoint

### Related pivotal tracker stories

	•[#163690184(https://www.pivotaltracker.com/story/show/163690184)

### Screenshots:

Screen Shot 2019-03-18 at 11 55 43

### CODE REVIEW

File: core/constants.py [hunk 1]
```diff
@@ -48,3 +48,17 @@
 YEAR_CHOICES = []
 for year in range(2013, (datetime.now().year + 1)):
     YEAR_CHOICES.append((year, year))
+
+NEWLY_REPORTED = "Newly reported"
+INTERNAL_ASSESSMENT = "Internel assessment"
+REQUIRES_REPAIR = "Requires repair"
```
Comment (MadgeKinyanjui) [inline core/constants.py]: Please view this story to see how state transitions should be implemented https://www.pivotaltracker.com/story/show/164099111

**Comment by nelmandela** [comment]
alright let me refactor that

File: core/models/asset.py [hunk 2]
```diff
@@ -558,3 +558,13 @@ def __str__(self):
 
     class Meta:
         ordering = ['-id']
+
+
+class StateTransitions(models.Model):
+    asset_incident_report = models.ForeignKey(
+        AssetIncidentReport, on_delete=models.PROTECT
+    )
+    transitions = models.CharField(max_length=50, default=constants.NEWLY_REPORTED)
```
Comment (egichuri) [inline core/models/asset.py]: let's call this `state` and restrict the choices

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,54 @@
+-i https://pypi.org/simple
```
Comment (egichuri) [inline requirements.txt]: please delete this since we have Pipfile

File: api/serializers/__init__.py [hunk 4]
```diff
@@ -22,6 +22,7 @@
     AssetStatusSerializer,
     AssetSubCategorySerializer,
     AssetTypeSerializer,
+    StateTransitionsSerializer,
```
Comment (egichuri) [inline api/serializers/__init__.py]: let's change this to `StateTransitionsSerializer`
Ideally the model, view, serializer names should be singular.

File: api/views/assets.py [hunk 5]
```diff
@@ -521,3 +522,10 @@ def get(self, request):
         response["Content-Disposition"] = 'attachment; filename="{}"'.format(filename)
 
         return response
+
+
+class StateTransitionsViewset(ModelViewSet):
```
Comment (egichuri) [inline api/views/assets.py]: this too

File: core/constants.py [hunk 6]
```diff
@@ -48,3 +48,5 @@
 YEAR_CHOICES = []
 for year in range(2013, (datetime.now().year + 1)):
     YEAR_CHOICES.append((year, year))
+
+NEWLY_REPORTED = "Newly reported"
```
Comment (egichuri) [inline core/constants.py]: You can add the other states as well

File: core/models/asset.py [hunk 7]
```diff
@@ -558,3 +558,13 @@ def __str__(self):
 
     class Meta:
         ordering = ['-id']
+
+
+class StateTransitions(models.Model):
```
Comment (egichuri) [inline core/models/asset.py]: this too

File: core/constants.py (hunk 6 above)
Comment (nelmandela) [reply core/constants.py]: that was to come in the next task, or i do it now?

Comment (egichuri) [reply core/constants.py]: If there's a task for it it's fine..

**Comment by nelmandela** [comment]
for new reports or existing ones?from my understanding of that is like keeping a history of incident reports, of which a new incident report doesn't have a history

**Comment by egichuri** [comment]
For new reports.. when a new incident report is submitted we should have an entry in the state transitions table

File: core/models/asset.py [hunk 8]
```diff
@@ -552,9 +552,36 @@ class AssetIncidentReport(models.Model):
     witnesses = models.TextField(null=True, blank=True)
     police_abstract_obtained = models.CharField(max_length=255)
     submitted_by = models.ForeignKey('User', null=True, on_delete=models.PROTECT)
+    state = models.CharField(max_length=50)
```
Comment (egichuri) [inline core/models/asset.py]: i think we can do without this field since the `StateTransition` model has a foreign key pointing to this table

**Comment by egichuri** [comment]
One last thing.. i see a couple of migration files.. please delete them and regenerate the migrations so that we'll have only one migrations file for this task.

**Pull request merged**